src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java

```diff
@@ -186,21 +189,9 @@ private void start() throws IOException {
             .start();
     logger.info(
         "Server started, listening on " + luceneServerConfiguration.getPort() + " for messages");
-
-    Runtime.getRuntime()
-        .addShutdownHook(
-            new Thread() {
-              @Override
-              public void run() {
-                // Use stderr here since the logger may have been reset by its JVM shutdown hook.
-                logger.error("*** shutting down gRPC server since JVM is shutting down");
-                LuceneServer.this.stop();
-                logger.error("*** server shut down");
-              }
-            });
   }
 
-  private void stop() {
+  public void stop() {
```

Comment:
nit: you could use the @VisibleForTesting to indicate this is widely visible just for testing purpose

Reply:
I don't see any harm if someone wants to import the server and start LuceneServer within their code, but still added it